pkg/plugin/v3/scaffolds/internal/templates/config/api/webhook_suitetest.go

```diff
@@ -0,0 +1,223 @@
+
+	// Only store code fragments in the map if the slices are non-empty
+	if len(addWebhookManager) != 0 {
+		fragments[file.NewMarkerFor(f.Path, addWebhookManagerMarker)] = addWebhookManager
+	}
+	if len(imports) != 0 {
+		fragments[file.NewMarkerFor(f.Path, importMarker)] = imports
+	}
+	if len(addScheme) != 0 {
+		fragments[file.NewMarkerFor(f.Path, addSchemeMarker)] = addScheme
+	}
+
+	return fragments
+}
+
+const (
+	webhookTestSuiteTemplate = `
+package {{ .Resource.Version }}
+
+import (
+	"path/filepath"
+	"testing"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	%s
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+// These tests use Ginkgo (BDD-style Go testing framework). Refer to
+// http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
+
+var cfg *rest.Config
+var k8sClient client.Client
+var testEnv *envtest.Environment
+var stopCh = make(chan struct{})
+
+func TestAPIs(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecsWithDefaultAndCustomReporters(t,
+	"Webhook Suite",
+	[]Reporter{printer.NewlineReporter{}})
+}
+
+var _ = BeforeSuite(func(done Done) {
+	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
+
+	By("bootstrapping test environment")
+	testEnv = &envtest.Environment{
+		CRDDirectoryPaths: []string{filepath.Join({{ .BaseDirectoryRelativePath }}, "config", "crd", "bases")},
+		WebhookInstallOptions: envtest.WebhookInstallOptions{
+			DirectoryPaths: []string{filepath.Join({{ .BaseDirectoryRelativePath }}, "config", "webhook")},
+		},
+	}
+
+	var err error
+	cfg, err = testEnv.Start()
+	Expect(err).ToNot(HaveOccurred())
+	Expect(cfg).ToNot(BeNil())
+
+	scheme := runtime.NewScheme()
+	err = AddToScheme(scheme)
+	Expect(err).NotTo(HaveOccurred())
+
+	%s
+	
+	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme})
+	Expect(err).ToNot(HaveOccurred())
+	Expect(k8sClient).ToNot(BeNil())
+
+	// start webhook server using Manager
+	wio := &testEnv.WebhookInstallOptions
+	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
+		Scheme:             scheme,
+		Host:               wio.LocalServingHost,
+		Port:               wio.LocalServingPort,
+		CertDir:            wio.LocalServingCertDir,
+		LeaderElection:     false,
+		MetricsBindAddress: "0",
+	})
+	Expect(err).ToNot(HaveOccurred())
+
+	%s
+
+	go func() {
+		err = mgr.Start(stopCh)
+		if err != nil {
+			panic(err)
+		}
+	}()
+
+	// wait for the webhook server to get ready
+	dialer := &net.Dialer{Timeout: time.Second}
+	addrPort := wio.LocalServingHost + ":" + fmt.Sprint(wio.LocalServingPort)
```

Comment:
consider using fmt.Sprintf

Reply:
+ for this suggestion

Reply:
Thank you for review. I tried to use the `fmt.Sprintf` in my code at the first attempt, but the scaffolding doesn't really add the %s in the scaffolded code. I initially tried to add the following line:
`addrPort := fmt.Sprintf("%s:%d", wio.LocalServingHost, wio.LocalServingPort)` but these %s in the scaffolded code tried to prints the following line after scaffolding:
`addrPort := fmt.Sprintf("%!s(MISSING):%!d(MISSING)", wio.LocalServingHost, wio.LocalServingPort)`. 

I tried to find the solution so that multiline string prints %s and %d as it is, but could not find that easily. Let me check again..

Reply:
@camilamacedo86 Do you know how to solve the above challenge I am facing?

Reply:
Did you try `%v`?

Reply:
Same error. We have to replace template body like this:
```
f.TemplateBody = fmt.Sprintf(webhookTestSuiteTemplate,
		file.NewMarkerFor(f.Path, importMarker),
		file.NewMarkerFor(f.Path, addSchemeMarker),
		file.NewMarkerFor(f.Path, addWebhookManagerMarker),
		"%v",
		"%v",
	)
```
So that, these %v can be replaced in the multi line string we have `webhookTestSuiteTemplate`

Reply:
Done